omx/gstomxh264dec.c

```diff
@@ -107,20 +107,125 @@ gst_omx_h264_dec_is_format_change (GstOMXVideoDec * dec,
       || g_strcmp0 (old_level, new_level) != 0) {
     return TRUE;
   }
-
   return FALSE;
 }
 
 static gboolean
 gst_omx_h264_dec_set_format (GstOMXVideoDec * dec, GstOMXPort * port,
     GstVideoCodecState * state)
 {
-  gboolean ret;
+  GstOMXH264Dec *self = GST_OMX_H264_DEC (dec);
+  GstCaps *peercaps;
   OMX_PARAM_PORTDEFINITIONTYPE port_def;
+  OMX_VIDEO_PARAM_PROFILELEVELTYPE param;
+  OMX_ERRORTYPE err;
+  gboolean ret;
+  const gchar *profile_string, *level_string;
 
   gst_omx_port_get_port_definition (port, &port_def);
   port_def.format.video.eCompressionFormat = OMX_VIDEO_CodingAVC;
   ret = gst_omx_port_update_port_definition (port, &port_def) == OMX_ErrorNone;
 
+  GST_OMX_INIT_STRUCT (&param);
+  param.nPortIndex = GST_OMX_VIDEO_DEC (self)->dec_in_port->index;
+
+  err =
+      gst_omx_component_get_parameter (GST_OMX_VIDEO_DEC (self)->dec,
+      OMX_IndexParamVideoProfileLevelCurrent, &param);
+  if (err != OMX_ErrorNone) {
+    GST_WARNING_OBJECT (self,
+        "Getting profile/level not supported by component");
+    return TRUE;
```

Comment:
This is an error, why are we returning TRUE?

Reply:
You are correct. b8a4d8c02b583e518e7574597b9f68abec87b778